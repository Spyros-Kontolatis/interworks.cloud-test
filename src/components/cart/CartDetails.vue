<template>
  <div class="border border-gray-400 p-4 rounded-md ml-4">
    <div class="flex flex-col">
      <h4 class="font-bold text-start" v-if="cart.items.length">Products:</h4>
      <div
        class="flex justify-between"
        v-for="{ product: { name, id, price }, quantity } in cart.items"
        :key="id"
      >
        <div class="flex grow">
          {{ name }}
          <div class="text-xs flex items-center ml-2">
            ({{ quantity }} x <PriceItem :price="price" />)
          </div>
        </div>
        <div class="text-xs flex gap-2">
          <PriceItem :price="price * quantity" />
        </div>
      </div>
      <div
        class="flex justify-between font-extrabold border-t border-t-gray-400 pt-2"
      >
        <div class="">TOTAL:</div>
        <div>
          <PriceItem :price="cart.totalPrice" />
        </div>
      </div>
    </div>
  </div>
</template>

<script setup lang="ts">
import { useCartStore } from "@/stores/cart";
import { storeToRefs } from "pinia";
import PriceItem from "../product/PriceItem.vue";

const cartStore = useCartStore();
const { cart } = storeToRefs(cartStore);
</script>
