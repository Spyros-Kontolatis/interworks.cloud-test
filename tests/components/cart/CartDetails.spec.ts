import { shallowMount } from "@vue/test-utils";
import CartDetails from "../../../src/components/cart/CartDetails.vue";

import { setActivePinia, createPinia } from "pinia";
import { createTestingPinia } from "@pinia/testing";

jest.mock("@/services/Service");
jest.mock("@/services/HTTP");
beforeEach(() => {
  setActivePinia(createPinia());
});
describe("CartDetails.vue", () => {
  it("renders the component with required props", () => {
    const wrapper = shallowMount(CartDetails);
    expect(wrapper.element).toMatchSnapshot();
  });
  it("correctly renders the details from the store", () => {
    const wrapper = shallowMount(CartDetails, {
      global: {
        plugins: [
          createTestingPinia({
            initialState: {
              cart: {
                cart: {
                  id: "1",
                  items: [
                    {
                      product: {
                        id: "1",
                        name: "Product 1",
                        price: 1,
                      },
                      quantity: 1,
                    },
                    {
                      product: {
                        id: "2",
                        name: "Product 2",
                        price: 2,
                      },
                      quantity: 2,
                    },
                  ],
                  totalPrice: 5,
                  totalQty: 3,
                },
              },
            },
          }),
        ],
      },
    });

    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.findAll(".cart-items")).toHaveLength(2);
  });
});
