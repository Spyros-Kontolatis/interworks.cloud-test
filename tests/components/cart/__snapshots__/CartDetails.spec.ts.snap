// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartDetails.vue correctly renders the details from the store 1`] = `
<div
  class="border border-gray-400 p-4 rounded-md ml-0 md:ml-4"
>
  <div
    class="flex flex-col"
  >
    <h4
      class="font-bold text-start"
    >
      Products:
    </h4>
    
    <div
      class="flex justify-between cart-items"
    >
      <div
        class="flex grow"
      >
        Product 1 
        <div
          class="text-xs flex items-center ml-2"
        >
           (1 x 
          <price-item-stub
            price="1"
          />
          ) 
        </div>
      </div>
      <div
        class="flex gap-2"
      >
        <price-item-stub
          price="1"
        />
      </div>
    </div>
    <div
      class="flex justify-between cart-items"
    >
      <div
        class="flex grow"
      >
        Product 2 
        <div
          class="text-xs flex items-center ml-2"
        >
           (2 x 
          <price-item-stub
            price="2"
          />
          ) 
        </div>
      </div>
      <div
        class="flex gap-2"
      >
        <price-item-stub
          price="4"
        />
      </div>
    </div>
    
    <div
      class="flex justify-between font-extrabold border-t border-t-gray-400 pt-2 totals"
    >
      <div
        class=""
      >
        TOTAL:
      </div>
      <div
        class="total-price"
      >
        <price-item-stub
          price="5"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`CartDetails.vue renders the component with required props 1`] = `
<div
  class="border border-gray-400 p-4 rounded-md ml-0 md:ml-4"
>
  <div
    class="flex flex-col"
  >
    <!--v-if-->
    
    
    <div
      class="flex justify-between font-extrabold border-t border-t-gray-400 pt-2 totals"
    >
      <div
        class=""
      >
        TOTAL:
      </div>
      <div
        class="total-price"
      >
        <price-item-stub
          price="0"
        />
      </div>
    </div>
  </div>
</div>
`;
